templates/cdk.master.leader.file-watcher.service

```diff
@@ -4,7 +4,9 @@ After=network.target
 
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/juju-run {{ unit }} /usr/local/sbin/cdk.master.leader.file-watcher.sh
+ExecStart=/bin/bash -c \
+    '/usr/bin/juju-exec {{ unit }} /usr/local/sbin/cdk.master.leader.file-watcher.sh || \
+    /usr/bin/juju-run {{ unit }} /usr/local/sbin/cdk.master.leader.file-watcher.sh'
```

Comment:
@addyess how about this to address your concern with both `juju-[exec|run]` failing?  kinda ugly, but `bash -c` was the simplest way i found to chain ExecStarts.

Confirmed with edge charms + this PR:
```
ubuntu@juju-8ac7ee-6:~$ systemctl status cdk.master.leader.file-watcher.service 
○ cdk.master.leader.file-watcher.service - CDK control-plane leader file-watcher
     Loaded: loaded (/etc/systemd/system/cdk.master.leader.file-watcher.service; disabled; vendor preset: enabled)
     Active: inactive (dead) since Mon 2023-02-13 18:02:50 UTC; 17s ago
TriggeredBy: ● cdk.master.leader.file-watcher.path
    Process: 1482911 ExecStart=/bin/bash -c /usr/bin/juju-exec kubernetes-control-plane/1 /usr/local/sbin/cdk.master.leader.file-watcher.sh ||      /usr/bin/juju-run kubernetes-contr>
   Main PID: 1482911 (code=exited, status=0/SUCCESS)
        CPU: 78ms

Feb 13 18:02:50 juju-8ac7ee-6 systemd[1]: Starting CDK control-plane leader file-watcher...
Feb 13 18:02:50 juju-8ac7ee-6 bash[1482912]: /bin/bash: line 1: /usr/bin/juju-exec: No such file or directory
Feb 13 18:02:50 juju-8ac7ee-6 systemd[1]: cdk.master.leader.file-watcher.service: Deactivated successfully.
Feb 13 18:02:50 juju-8ac7ee-6 systemd[1]: Finished CDK control-plane leader file-watcher.
```

Reply:
Welp, this isn't holding up much better; looks like we're going for the 6hr timeout on tests.

Reply:
this wasn't the reason behind the test-timeouts.  Turns out i had the wrong version of python libjuju selected in tox.ini.  I should have constrained it to `juju < 3.1`.  The 3.0 version of `juju` library has improved async handling on run events -- the very first `juju_run` call in the test was the point of the failure